Gems/AtomLyIntegration/CommonFeatures/Code/Source/Mesh/MeshComponentController.cpp

```diff
@@ -287,6 +295,11 @@ namespace AZ
             {
                 m_meshFeatureProcessor->SetTransform(m_meshHandle, world, m_cachedNonUniformScale);
             }
+
+            // ensure the render geometry is kept in sync with any changes to the entity the mesh is on
+            AzFramework::RenderGeometry::IntersectionNotificationBus::Event(
```

Comment:
Why not have the intersector listen to TransformNotificationBus::OnTransformChanged and dirty the entity whenever the transform changes? Otherwise every component that handles the IntersectionRequestBus needs to know to put this call in OnTransformChanged.